math/cosine_test.go

```diff
@@ -1,176 +1,73 @@
+		{neg(_2pi, N), "1.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"},
+	} {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			z := decimal.WithPrecision(N)
+			x, _ := new(decimal.Big).SetString(tt.x)
+			r, _ := new(decimal.Big).SetString(tt.r)
+
+			Cos(z, x)
+			if z.Cmp(r) != 0 && diff.Sub(r, z).CmpAbs(eps) > 0 {
+				t.Errorf(`#%d: Cos(%s)
+wanted: %s
+got   : %s
+diff  : %s
+`, i, x, r, z, diff)
 			}
 		})
 	}
 }
 
-func BenchmarkCosZero(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(strZero, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCosPiOver6(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(strPiOver6, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCosPiOver5(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(strPiOver5, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCosPiOver4(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(strPiOver4, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCosPiOver3(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(strPiOver3, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCosPiOver2(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(strPiOver2, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCos2PiOver3(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(str2PiOver3, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCosPi(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(strPi, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
-	}
-}
-
-func BenchmarkCos2Pi(b *testing.B) {
-	precision := 30
-	theta := newDecimal(valuePosStr(str2Pi, uint(precision)))
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Cos(result, theta)
+func BenchmarkCos(b *testing.B) {
+	x := new(decimal.Big)
+	for i, n := range [...]string{
+		"0", _pi_6, _pi_5, _pi_4, _pi_3, _pi_2, _2pi_3, _pi, _2pi,
+	} {
+		b.Run(fmt.Sprintf("%d", i), func(b *testing.B) {
+			for _, prec := range benchPrecs {
+				x.SetString(pos(n, uint(prec)))
```

Comment:
This will fail because "0" does not have enough chars, and because benchPrecs has values>1000 for the others

Reply:
Oh, yeah. Maybe I’ll remove the call to `pos` or something.